### PR TITLE
feat(TBD-10921):Add Spark local 3.x

### PR DESCRIPTION
**Pull request by lbourgeois**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

**Comment by lbourgeois** [comment]
build-patch

File: main/plugins/org.talend.hadoop.distribution.spl30x/plugin.xml [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+             id="scala-library-2.12.10.jar" name="scala-library-2.12.10.jar"
+             mvn_uri="mvn:org.scala-lang/scala-library/2.12.10"/>
+
+         <libraryNeeded context="plugin:org.talend.hadoop.distribution.spl30x"
+             id="commons-lang3-3.9.jar" name="commons-lang3-3.9.jar"
+             mvn_uri="mvn:org.apache.commons/commons-lang3/3.9"/>
+
+         <libraryNeeded context="plugin:org.talend.hadoop.distribution.spl30x"
+             id="scala-library-2.12.10.jar" name="scala-library-2.12.10.jar"
+             mvn_uri="mvn:org.scala-lang/scala-library/2.12.10"/>
```
Comment (msjian) [inline main/plugins/org.talend.hadoop.distribution.spl30x/plugin.xml:73]: this is a duplicate part
<libraryNeeded context="plugin:org.talend.hadoop.distribution.spl30x"
             id="scala-library-2.12.10.jar" name="scala-library-2.12.10.jar"
             mvn_uri="mvn:org.scala-lang/scala-library/2.12.10"/>

Comment (lbourgeois) [reply main/plugins/org.talend.hadoop.distribution.spl30x/plugin.xml:73]: Good catch, will clean. Thanks

**Comment by lbourgeois** [comment]
build-patch

**Comment by msjian** [comment]
build-patch

**Comment by lbourgeois** [comment]
build-patch

**Comment by lbourgeois** [comment]
build-patch

**Review by cbadillo1603 — DISMISSED** [review]
reviewed

**Comment by lbourgeois** [comment]
build-patch

**Review by AlixMetivier — DISMISSED** [review]

**Review by cbadillo1603 — DISMISSED** [review]
reviewed

**Comment by lbourgeois** [comment]
build-patch

**Comment by lbourgeois** [comment]
build-patch

**Review by cbadillo1603 — APPROVED** [review]
reviewed

**Pull request merged**